st2common/st2common/models/api/action.py

```diff
@@ -148,6 +149,11 @@ class ActionAPI(BaseAPI):
                 "patternProperties": {
                     "^\w+$": util_schema.get_draft_schema()
                 }
+            },
+            "tags": {
+                "description": "User associated metadata assigned to this object.",
```

Comment:
I am partial to tags mostly because it is something we constantly referred to in internal discussion and also a term used in previous APIs I worked with. Aesthetically I prefer tags because its shorter than metadata and can be verb-ified to 'tag an object' etc. However, I will concede that metadata is the more accurate terms and tags is well just the slightly more pleasing one. 

I would let @dzimine @lakshmi-kannan @DoriftoShoes @jfryman @m4dcoder @enykeev chime in and put forth their preference so as to end up with a pseudo-democratic winner :).